crates/pallets/starknet/src/lib.rs

```diff
@@ -1087,7 +1087,7 @@ impl<T: Config> Pallet<T> {
         )?;
 
         let mut results = vec![];
-        for res in execution_results {
+        for (res, _) in execution_results {
```

Comment:
The fact it is not used here mean we are doing extra work for nothing.
Create another method: `execute_txs_and_rollback_with_state_diff`. Or make it return `Option<CommitmentStateDiff>` and dynamically decide which one to use

Reply:
Done, went for the 2nd approach to avoid duplication